### ISSUE TITLE
Using fmu_instance argument is causing exception in fmi2SetupExperiment

### ISSUE THREAD

**Issue by apanys** [CATIA-Systems/FMPy#182]
Please open attached example in Visual Studio Code with extension `ms-vscode-remote.remote-containers` enabled.
Run the sample in Visual Studio Code terminal using command `python sample.py`.
[sample.zip](https://github.com/CATIA-Systems/FMPy/files/5438618/sample.zip)

``` python
import numpy as np
import pandas as pd
from fmpy import read_model_description, extract
from fmpy.simulation import instantiate_fmu
from fmpy.simulation import simulate_fmu


def create_processor(fmu_file_path):
    unzipdir = extract(fmu_file_path)
    model_description = read_model_description(fmu_file_path, validate=False)
    fmu = instantiate_fmu(unzipdir, model_description, fmi_type=None, visible=False,
                          debug_logging=False, logger=None, fmi_call_logger=None,
                          use_remoting=False)

    def process(*args, step_size=None):
        df_in = args[0]
        print(df_in)
        start_time = df_in.index[0]
        stop_time = df_in.index[-1]

        df_in = pd.DataFrame.to_records(df_in)
        df_out = simulate_fmu(fmu_file_path, validate=False, input=df_in, step_size=step_size,
                              start_values={}, output_interval=1, start_time=start_time,
                              stop_time=stop_time, debug_logging=False, fmu_instance=fmu)
        return pd.DataFrame.from_records(df_out).set_index('time')
    return process

def main():
    df_window1 = pd.DataFrame(data=np.array(
        [[0.0, 0.0], [1.0, 0.0]], np.double), columns=['time', 'myInput']).set_index('time')
    df_window2 = pd.DataFrame(data=np.array(
        [[2.0, 1.718294], [3.0, 8.107423]], np.double), columns=['time', 'myInput']).set_index('time')

    processor = create_processor('parametersModel.fmu')
    processor(df_window1, 1)
    processor(df_window2, 1)


if __name__ == '__main__':
    main()
```

Argument `fmu_instance` is used in function `simulate_fmu` to process 2 windows. On second window, exception
is thrown:

> Traceback (most recent call last):
  File "sample.py", line 40, in <module>
    main()
  File "sample.py", line 36, in main
    processor(df_window2, 1)
  File "sample.py", line 24, in process
    stop_time=stop_time, debug_logging=False, fmu_instance=fmu)
  File "/usr/local/lib/python3.7/site-packages/fmpy/simulation.py", line 571, in simulate_fmu
    result = simulateME(model_description, fmu, start_time, stop_time, solver, step_size, relative_tolerance, start_values, apply_default_start_values, input, output, output_interval, record_events, timeout, step_finished)
  File "/usr/local/lib/python3.7/site-packages/fmpy/simulation.py", line 697, in simulateME
    fmu.setupExperiment(startTime=start_time)
  File "/usr/local/lib/python3.7/site-packages/fmpy/fmi2.py", line 280, in setupExperiment
    return self.fmi2SetupExperiment(self.component, toleranceDefined, tolerance, startTime, stopTimeDefined, stopTime)
  File "/usr/local/lib/python3.7/site-packages/fmpy/fmi2.py", line 215, in w
    raise Exception("%s failed with status %d." % (fname, res))
Exception: fmi2SetupExperiment failed with status 3.

FMU is built using OMC and based on this model description:

```
model parametersModel
	//Interface(s)
	Modelica.Blocks.Interfaces.RealInput myInput;
	Modelica.Blocks.Interfaces.RealOutput myOutput;

	//Declaration(s)
	parameter Real myParameter = 1.0;
	Real x;
	//Component(s)
	Modelica.Blocks.Sources.RealExpression Real1 (y = x);


equation
	der(x) = myParameter*x + myInput;
	//Connection(s)
	connect(myOutput, Real1.y);
end  parametersModel;
```

**Comment by t-sommer** [comment]
This seems to be a bug in the FMU. You might want to try a newer version of Open Modelica.

**Comment by apanys** [comment]
I just created a new FMU using latest released version of OpenModelica (1.16.0-1). Unfortunately the same error is thrown.
[parametersModel_1.16.0-1.zip](https://github.com/CATIA-Systems/FMPy/files/5444234/parametersModel_1.16.0-1.zip)

**Comment by t-sommer** [comment]
The solution proposed in #183 should also work for this model. If you want to reuse the FMU instance you can use https://github.com/CATIA-Systems/FMPy/blob/master/fmpy/examples/efficient_loops.py as a starting point. Also note that you have to reset the FMU and clean up before you return.
